processing/src/main/java/io/druid/segment/incremental/OffheapIncrementalIndex.java

```diff
@@ -71,37 +84,61 @@
 
   private String outOfRowsReason = null;
 
+  private final File tmpDirLocation;
+
   public OffheapIncrementalIndex(
       IncrementalIndexSchema incrementalIndexSchema,
       boolean deserializeComplexMetrics,
       boolean reportParseExceptions,
       boolean sortFacts,
-      int maxRowCount,
-      StupidPool<ByteBuffer> bufferPool
+      long maxOffheapSize
   )
   {
     super(incrementalIndexSchema, deserializeComplexMetrics, reportParseExceptions, sortFacts);
-    this.maxRowCount = maxRowCount;
-    this.bufferPool = bufferPool;
+    tmpDirLocation = Files.createTempDir();
+
+    try {
+      File mapDbFile = File.createTempFile("druid-offheap-index-mapdb", "", tmpDirLocation);
+      File aggsBBFile = File.createTempFile("druid-offheap-index-aggs", "", tmpDirLocation);
+
+      AtomicLong availableTotalSpace = new AtomicLong(maxOffheapSize);
+
+      final DBMaker dbMaker = new MyDBMaker(mapDbFile, 24, 29, availableTotalSpace)
+          .deleteFilesAfterClose()
+          .commitFileSyncDisable()
+          .cacheSize(1024)
```

Comment:
yes, but then this impl is supposed to be "off-heap", increasing cache size means you are using more and more of heap. i debated to use cache at all but then decided to use very little so that if your input data was , say sorted, this would help.